src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,474 @@
+     * @param rangePayments The range payments being created
+     */
+    function payAllForRange(
+        RangePayment[] calldata rangePayments
+    ) external onlyWhenNotPaused(PAUSED_PAY_ALL_FOR_RANGE) onlyPayAllForRangeSubmitter nonReentrant {
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            uint256 nonce = paymentNonce[msg.sender];
+            bytes32 rangePaymentForAllHash = keccak256(abi.encode(msg.sender, nonce, rangePayment));
+            require(
+                !isRangePaymentForAllHash[msg.sender][rangePaymentForAllHash],
+                "PaymentCoordinator._payForRange: range payment hash already submitted"
+            );
+            isRangePaymentForAllHash[msg.sender][rangePaymentForAllHash] = true;
+
+            _payForRange(rangePayment);
+            emit RangePaymentForAllCreated(msg.sender, nonce, rangePaymentForAllHash, rangePayment);
+            paymentNonce[msg.sender] = nonce + 1;
+        }
+    }
+
+    /**
+     * @notice Claim payments against a given root (read from distributionRoots[claim.rootIndex]).
+     * Earnings are cumulative so earners don't have to claim against all distribution roots they have earnings for,
+     * they can simply claim against the latest root and the contract will calculate the difference between
+     * their cumulativeEarnings and cumulativeClaimed. This difference is then transferred to claimerFor[claim.earner]
+     * @param claim The PaymentMerkleClaim to be processed.
+     * Contains the root index, earner, payment leaves, and required proofs
+     * @dev only callable by the valid claimer, that is
+     * if claimerFor[claim.earner] is address(0) then only the earner can claim, otherwise only
+     * claimerFor[claim.earner] can claim the payments.
+     */
+    function processClaim(PaymentMerkleClaim calldata claim) external onlyWhenNotPaused(PAUSED_CLAIM_PAYMENTS) nonReentrant {
+        DistributionRoot memory root = distributionRoots[claim.rootIndex];
+        require(_checkClaim(claim, root), "PaymentCoordinator.processClaim: claim does not pass the check");
+        // If claimerFor earner is not set, claimer is by default the earner. Else set to claimerFor
+        address earner = claim.earnerLeaf.earner;
+        address claimer = claimerFor[earner];
+        if (claimer == address(0)) {
+            claimer = earner;
+        }
+        require(msg.sender == claimer, "PaymentCoordinator.processClaim: caller is not valid claimer");
+        for (uint256 i = 0; i < claim.tokenIndices.length; i++) {
+            TokenTreeMerkleLeaf calldata tokenLeaf = claim.tokenLeaves[i];
+
+            // Calculate amount to claim and update cumulativeClaimed
+            // Will revert if new leaf cumulativeEarnings is less than cumulative claimed
+            uint256 claimAmount = tokenLeaf.cumulativeEarnings - cumulativeClaimed[earner][tokenLeaf.token];
```

Comment:
Doing an explicit check/require for the underflow case here will make it easier to write tests!

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/24fb70726a39a125e61cc19a361d7cc20aa7b930
Added